### DISCUSSION TITLE
Difference between 'atBottomStateChange' and 'endReached'

### DISCUSSION THREAD

**Discussion by fabriciodorneles** [petyosi/react-virtuoso#372] [Q&A]
Hello all!

There is a difference between this 2 props, besides the different returns? I´m using this to make call to my API to load more items on the list.
when you say that is 'called with true / false when the list has reached the bottom / gets scrolled up. ' The only way to trigger this is by scrolling, or you can trigger in other way, like filling items in list, or changing an index?

There is a link to a good advanced tutorial to Virtuoso? I could not find some good advanced material.

Thank you!

**Comment by petyosi** [comment]
The `bottomStateChange` is a convenience API which lets you toggle 'scroll to bottom' buttons, common in chat UI - [see example](https://virtuoso.dev/stick-to-bottom/). 
The `endReached` does what you need. It gets called exactly once for a certain list size (or data set), so you don't have to guard against user scrolling up/down while you load - [see example](https://virtuoso.dev/endless-scrolling/). 

Re: advanced tutorial: I would not describe any of the examples available on the website as an advanced tutorial. While somewhat opinionated, the library is fairly flexible. What are you trying to achieve?

**Reply by fabriciodorneles** [reply]
Thanks for the response!
I need to use virtuoso in a complex way. My API a list with hundreds of thousands items.
And when the user select one item in anywhere of this list, I need to load the page(50 or 100 items by page) that contains the item, and position the item at the top of the list.
Also, when the user scroll up ou down to the limits of the list I need to load the respective pages.
The loading in the selection is OK, but my problem is when I use the method 'scrollToIndex' to position the item at the top, virtuoso is rendering the upward page, like the list have scrolled up to the limits. This happens with the initial page items. 
I´m defining the 'firstItemIndex' with the length of the list initially loaded when the selection is made.

**Reply by fabriciodorneles** [reply]
I managed to make it work. We have made here a confusion with the use of the 'firstItemIndex' and 'totalCount'. 
This part of the documentation clarified for me: https://virtuoso.dev/prepend-items/.

Thank you for the reponse and for the work with Virtuoso. It´s an awesome tool!

**Reply by fabriciodorneles** [reply]
Sometimes, rarely, it goes wrong. I load the list, and position on the item, and the previous pages loads(and this should not happen). I think maybe there is some kind of delay in the list render that causes the scrollToIndex() function happens before the complete render of the list. But I´m not sure of that. But this is happen very rarely now.
